nodecg-io-curseforge/extension/curseforgeClient.ts

```diff
@@ -0,0 +1,1034 @@
+        });
+
+        return new Date(await response.text());
+    }
+
+    /**
+     * Get the UTC time when the minecraft versions were last updated.
+     */
+    async getMinecraftVersionTimestamp(): Promise<Date> {
+        const response = await fetch("https://addons-ecs.forgesvc.net/api/v2/minecraft/version/timestamp", {
+            method: "GET",
+        });
+
+        return new Date(await response.text());
+    }
+
+    /**
+     * Get the UTC time when the mod loader list was last updated.
+     */
+    async getModLoaderTimestamp(): Promise<Date> {
+        const response = await fetch("https://addons-ecs.forgesvc.net/api/v2/minecraft/modloader/timestamp", {
+            method: "GET",
+        });
+
+        return new Date(await response.text());
+    }
+
+    /**
+     * Get the UTC time when the categories were last updated.
+     */
+    async getCategoryTimestamp(): Promise<Date> {
+        const response = await fetch("https://addons-ecs.forgesvc.net/api/v2/category/timestamp", {
+            method: "GET",
+        });
+
+        return new Date(await response.text());
+    }
+
+    /**
+     * Get the UTC time when the games were last updated.
+     */
+    async getGameTimestamp(): Promise<Date> {
+        const response = await fetch("https://addons-ecs.forgesvc.net/api/v2/game/timestamp", {
+            method: "GET",
+        });
+
+        return new Date(await response.text());
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+    //                                                                                                       //
+    //   METHODS FROM HERE ONWARDS UNTIL END OF CLASS CurseForge ARE NOT MEANT TO BE CALLED BY BUNDLES.      //
+    //   THEY MAY GIVE MORE POSSIBILITIES BUT YOU CAN ALSO BREAK MUCH WITH IT. CALL THEM AT YOUR OWN RISK.   //
+    //                                                                                                       //
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    // eslint-disable-next-line
+    async rawRequest(method: HttpMethod, endpoint: string, data?: any): Promise<any> {
+        const response = await fetch(`https://addons-ecs.forgesvc.net/api/v2/${endpoint}`, {
+            method: method,
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
+            },
+            body: data == undefined ? undefined : JSON.stringify(data),
+        });
+
+        return response.json();
+    }
+}
+
+/**
+ * A curse addon such as a Minecraft mod.
+ */
+export class CurseAddon {
+    private readonly curse: CurseForge;
+    readonly addonId: number;
+    readonly info: CurseAddonInfo;
+
+    constructor(curse: CurseForge, addonId: number, info: CurseAddonInfo) {
+        this.curse = curse;
+        this.addonId = addonId;
+        this.info = info;
+    }
```

Comment:
nit: You can declare class members right in the constructor parameter list to make this simpler:
```suggestion
    constructor(private readonly curse: CurseForge, public readonly addonId: number, public readonly info: CurseAddonInfo) {}
```